MatrixSDK/Data/MXRoom.m

```diff
@@ -779,21 +800,24 @@ - (MXHTTPOperation*)setIsDirect:(BOOL)isDirect
         NSString *directUserId = [self getDirectUserId];
         if (directUserId)
         {
-            NSMutableDictionary *directRooms = [NSMutableDictionary dictionaryWithDictionary:mxSession.directRooms];
-            NSMutableArray *roomLists = [NSMutableArray arrayWithArray:directRooms[directUserId]];
-            
-            [roomLists removeObject:self.roomId];
-            
-            if (roomLists.count)
-            {
-                [directRooms setObject:roomLists forKey:directUserId];
-            }
-            else
+            @synchronized (mxSession.directRooms)
```

Comment:
In the sdk, we avoid to use @synchronized as the contract is to call the SDK API from the UI thread.
If we start doing thread locks, we will need to do it for every SDK methods